monitoring_tools.py

```diff
@@ -41,31 +41,36 @@ def check_config_present(self):
                 raise ValueError(f"Config is missing term '{term}'")
 
     def add_imagepaths_to_html(self, image_dict):
-        """add_filepath_to_monitor_txt: adds single line to end of txt file,
-            in this case with 4 spaces,
-            to keep alignment with generic template
-            args:
-                image_path: path to image file added
-                id: the id associated with that image
-                success: indicates whether image at filepath failed to upload to image server or not
+        """add_imagepaths_to_html: adds an image path row to the HTML table before the closing </table> tag.
+        Args:
+            image_dict: dictionary where the key is the image ID and value is a list of tuples
+                        containing image paths and their success status.
         """
         for key, value in image_dict.items():
             img_id = key
-            image_path = value[0]
-            success = value[1]
-
-            monitor_line = f"<tr style='width: 50%'><td>{image_path}</td> <td>{img_id}</td><td>{success}</td></tr>"
-
-            with open(self.path, "r") as file:
-                html_content = file.readlines()
-
-            insert_position = len(html_content) - 6
-
-            html_content.insert(insert_position, monitor_line + '\n')
-
-            # Write the updated HTML content back to the file
-            with open(self.path, 'w') as file:
-                file.writelines(html_content)
+            for result in value:
+                image_path = result[0]
+                success = result[1]
+                monitor_line = f"<tr style='width: 50%'><td>{image_path}</td> <td>{img_id}</td><td>{success}</td></tr>"
+
+                with open(self.path, "r") as file:
+                    html_content = file.readlines()
+
+                insert_position = None
+
+                # Find the position of the closing </table> tag
+                for i, line in enumerate(html_content):
```

Comment:
now uses objective position relative to table to insert line rather than hardcoding the line number